internal/services/springcloud/spring_cloud_api_portal_resource.go

```diff
@@ -4,290 +4,379 @@
+			if sku == nil {
+				return fmt.Errorf("retrieving %s: sku was nil", id)
+			}
+
+			if metadata.ResourceData.HasChange("gateway_ids") {
+				properties.GatewayIds = pointer.To(model.GatewayIds)
+			}
+
+			if metadata.ResourceData.HasChange("https_only_enabled") {
+				properties.HTTPSOnly = pointer.To(model.HttpsOnlyEnabled)
+			}
+
+			if metadata.ResourceData.HasChange("instance_count") {
+				sku.Capacity = pointer.To(int64(model.InstanceCount))
+			}
+
+			if metadata.ResourceData.HasChange("public_network_access_enabled") {
+				properties.Public = pointer.To(model.PublicNetworkAccessEnabled)
+			}
+
+			if metadata.ResourceData.HasChange("sso") {
+				properties.SsoProperties = expandApiPortalSsoProperties(model.Sso)
+			}
+
+			resource := appplatform.ApiPortalResource{
+				Properties: properties,
+				Sku:        sku,
+			}
+
+			err = client.ApiPortalsCreateOrUpdateThenPoll(ctx, *id, resource)
+			if err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.ApiPortalName)
-	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
-			log.Printf("[INFO] appplatform %q does not exist - removing from state", d.Id())
-			d.SetId("")
 			return nil
-		}
-		return fmt.Errorf("retrieving %s: %+v", id, err)
-	}
-	d.Set("name", id.ApiPortalName)
-	d.Set("spring_cloud_service_id", parse.NewSpringCloudServiceID(id.SubscriptionId, id.ResourceGroup, id.SpringName).ID())
-	if resp.Sku != nil {
-		d.Set("instance_count", resp.Sku.Capacity)
-	}
-	if props := resp.Properties; props != nil {
-		d.Set("gateway_ids", flattenSpringCloudAPIPortalGatewayIds(props.GatewayIds))
-		d.Set("https_only_enabled", props.HTTPSOnly)
-		d.Set("public_network_access_enabled", props.Public)
-		if err := d.Set("sso", flattenAPIPortalSsoProperties(props.SsoProperties, d.Get("sso").([]interface{}))); err != nil {
-			return fmt.Errorf("setting `sso`: %+v", err)
-		}
-		d.Set("url", props.URL)
+		},
 	}
-	return nil
 }
 
-func resourceSpringCloudAPIPortalDelete(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).AppPlatform.APIPortalClient
-	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
-	defer cancel()
+func (s SpringCloudAPIPortalResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AppPlatform.AppPlatformClient
 
-	id, err := parse.SpringCloudAPIPortalID(d.Id())
-	if err != nil {
-		return err
-	}
+			id, err := appplatform.ParseApiPortalID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.ApiPortalsGet(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+			if resp.Model == nil {
```

Comment:
For read, we shall not error out when `resp.Model` is nil, see: https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-new-resource.md#:~:text=//%20the%20SDK%20will,!%3D%20nil%20%7B